subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultFilePropertyFactory.java

```diff
@@ -116,17 +117,18 @@ public File getAsFile() {
         }
     }
 
-    static class ResolvingFile extends AbstractMappingProvider<RegularFile, CharSequence> implements TaskDependencyContainer {
+    static class ResolvingFile extends AbstractMappingProvider<RegularFile, CharSequence> {
```

Comment:
What is this? It would be useful if this type has some sort of documentation to describe what a `ResolvingFile` actually is.